### PR TITLE
i#2868 pie/pic for native takeover: force plt stubs

### PR DESCRIPTION
**Pull request by hgreving2304**
In native executable to DR controlled dll takeover test, we need PLT stubs to work. This is adding no-pie option for these tests, as some newer unix distrubutions enable pie/pic by default.

Issue: #2868

### CODE REVIEW

**Comment by hgreving2304** [comment]
Apparently travis is running with gcc 4.8 that doesn't know the -no-pie option (or the option is only enable when default pie is turned on)

File: suite/tests/CMakeLists.txt [hunk 1]
```diff
@@ -547,6 +548,19 @@ function(add_exe test source)
     # it hard to use other DR builds w/ these apps
     SKIP_BUILD_RPATH ON)
   if (UNIX)
+    # for common.nativeexec takeover tests, we do not support -pie/pic. newer (unix-) distributions
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: style violations here and below: line too long

Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: style violations here and below: multi-sentence comment must use capitilization and end punctuation (also preferred for single-sentence)

Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: I find "pie/pic" confusing: "pie" is clear as it means the exe itself is PIC and ET_DYN; "pic" without saying that it's applied to the exe makes me think we don't support .so files or sthg.  I would just remove the "/pic" as it does not add anything.

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -547,6 +548,19 @@ function(add_exe test source)
...
+    # with toolchains that enabled pie/pic by default. to takeover the "PLT style" dll call (dll not
+    # native, exe in native list), we expect the call to jump to a PLT stub first. however with
+    # pie/pic, the compiler will generate a rip-relative gotplt table call instead. we do not have
+    # support to takeover such calls AFAIK
+    CHECK_C_COMPILER_FLAG("-no-pie" no_pie_avail)
+    if (no_pie_avail)
+      set_source_files_properties(common/nativeexec.c PROPERTIES COMPILE_FLAGS "${CMAKE_C_FLAGS} -fno-pie")
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: style: line too long: below also

File: suite/tests/CMakeLists.txt [hunk 3]
```diff
@@ -547,6 +548,19 @@ function(add_exe test source)
...
+      if ("${test}" MATCHES "common.nativeexec")
+        get_target_property(cur_ldflags ${test} LINK_FLAGS)
+        set_target_properties(common.nativeexec PROPERTIES LINK_FLAGS "-no-pie ${cur_ldflags}")
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: use append_link_flags()

File: suite/tests/CMakeLists.txt [hunk 4]
```diff
@@ -536,6 +536,7 @@ function(add_exe test source)
   endif ("${srccode}" MATCHES "include \"test.*annotation.*.h\"")
 
   add_executable(${test} ${test_srcs})
+
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: revert

File: suite/tests/CMakeLists.txt (hunk 1 above)
Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: Fixed

File: suite/tests/CMakeLists.txt [hunk 5]
```diff
@@ -548,17 +547,17 @@ function(add_exe test source)
...
     SKIP_BUILD_RPATH ON)
   if (UNIX)
-    # for common.nativeexec takeover tests, we do not support -pie/pic. newer (unix-) distributions
-    # with toolchains that enabled pie/pic by default. to takeover the "PLT style" dll call (dll not
-    # native, exe in native list), we expect the call to jump to a PLT stub first. however with
-    # pie/pic, the compiler will generate a rip-relative gotplt table call instead. we do not have
-    # support to takeover such calls AFAIK
+    # For common.nativeexec takeover tests, we do not support PIE. Newer (unix-) distri-
+    # butions with toolchains may enable PIE by default. In order to be able take over the
+    # "PLT style" dll call (test with excutablee in native exec list), we expect the call
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: typo: "executablee"

Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: Fixed

**Pull request merged**